src/data_snack/entities/entity_meta.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABCMeta, ABC
+from typing import List, get_type_hints
+
+from data_snack.entities.exceptions import MetaFieldsException, MetaEmptyKeysException
+
+
+class EntityMetaClass(ABCMeta):
+
+    class Meta:
```

Comment:
I don't think we should have two `Meta` class definitions. Instead I would store a dict of fields with types here.